File: tests/fixtures/polyglot/sample.go
// Polyglot fixture — unchecked error return.
// `os.Open` returns (*File, error). Line 10 drops err with `_`. File may be nil.
// Parses clean; staticcheck / errcheck would flag line 10.

package main

import "os"

func readConfig(path string) []byte {
	f, _ := os.Open(path)
	buf := make([]byte, 64)
	f.Read(buf)
	return buf
}

func main() {
	readConfig("/nonexistent")
}

File: tests/fixtures/polyglot/sample.rs
// Polyglot fixture — Option::unwrap() panic path.
// `head()` returns Option<i32>; line 10 unwraps on an empty vec.
// Parses clean; clippy::unwrap_used would flag line 10.

fn head(xs: &Vec<i32>) -> Option<i32> {
    xs.first().copied()
}

fn main() {
    let empty: Vec<i32> = Vec::new();
    let v = head(&empty).unwrap();
    println!("{}", v);
}
